Examples/Algorithms/Digitization/src/SmearingAlgorithm.cpp

```diff
@@ -8,85 +8,146 @@
 
 #include "ActsExamples/Digitization/SmearingAlgorithm.hpp"
 
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Utilities/ParameterDefinitions.hpp"
 #include "ActsExamples/EventData/GeometryContainers.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/Measurement.hpp"
 #include "ActsExamples/EventData/SimHit.hpp"
 #include "ActsExamples/Framework/WhiteBoard.hpp"
-#include <Acts/Geometry/GeometryIdentifier.hpp>
-#include <Acts/Geometry/TrackingGeometry.hpp>
+#include "ActsFatras/Digitization/UncorrelatedHitSmearer.hpp"
 
 #include <stdexcept>
 
 ActsExamples::SmearingAlgorithm::SmearingAlgorithm(
     ActsExamples::SmearingAlgorithm::Config cfg, Acts::Logging::Level lvl)
     : ActsExamples::BareAlgorithm("SmearingAlgorithm", lvl),
       m_cfg(std::move(cfg)) {
-  if (m_cfg.inputSimulatedHits.empty()) {
-    throw std::invalid_argument("Missing input hits collection");
+  if (m_cfg.inputSimHits.empty()) {
+    throw std::invalid_argument("Missing simulated hits input collection");
   }
   if (m_cfg.outputMeasurements.empty()) {
-    throw std::invalid_argument("Missing output measurement collection");
+    throw std::invalid_argument("Missing measurements output collection");
   }
-  if (!m_cfg.randomNumbers) {
+  if (m_cfg.outputSourceLinks.empty()) {
+    throw std::invalid_argument("Missing source links output collection");
+  }
+  if (m_cfg.outputMeasurementParticlesMap.empty()) {
```

Comment:
Potential problem here:
- imagine we want to add noise as random sim hits w/o output measurement particle map, will that be safe?

Reply:
That should not be a problem. The map does not require entries for every generated measurements. Only for the once where we know the association. This should also work with the ATLAS ITk datasets where a the pileup hits have no associated particle.